src/sentry/dynamic_sampling/rules/biases/boost_environments_bias.py

```diff
@@ -45,6 +45,37 @@ def _generate_bias_rules(self, bias_data: BiasData) -> List[BaseRule]:
         ]
 
 
+class BoostEnvironmentsRulesGeneratorV2(BiasRulesGenerator):
+    def _generate_bias_rules(self, bias_data: BiasData) -> List[PolymorphicRule]:
+        return [
+            {
+                "samplingValue": {
+                    "type": "sampleRate",
+                    "value": 1.0,
+                },
+                "type": "trace",
+                "condition": {
+                    "op": "or",
+                    "inner": [
+                        {
+                            "op": "glob",
+                            "name": "trace.environment",
+                            "value": ENVIRONMENT_GLOBS,
+                            "options": {"ignoreCase": True},
+                        }
+                    ],
+                },
+                "active": True,
+                "id": bias_data["id"],
+            }
+        ]
```

Comment:
There is a lot of code duplication in these generators, which could be prevented by calling the V1 generator from the V2 generator and then translating the result to the new format. But since this double generation will only exist for a short period of time, I think it's fine to leave it as-is.

Reply:
The idea was actually to simply duplicate because we are going to fade it out.

Reply:
agree with Joris, not sure why do we want to copy everything. It's okay for a temp solution. But it makes me think our current framework has some caveats and especially it means we can't extend easily 🤔

Reply:
It was super easy to extend, nothing difficult. The repetition has been kept just because we need both active, this is an edge case.

There are many ways of solving this problem but I prefer how explicit this solution is, especially keeping in mind that we are going to remove the old format.